### PR TITLE
api: Add http api and deprecate rpc api

### PR DESCRIPTION
**Pull request by hyochan-brown**
## Proposed changes

To enhance compatibility with Ethereum.
- Rename node/utils.go to node/api.go.  
   It feels awkward to have an api in the utils.go. It is also not compatible with Ethereum.
- Add Start/StopHTTP & Deprecate Start/StopRPC
- Add test code for http/ws api
   I wrote it by referring to the test code of Ethereum.
   
The referenced Ethereum [PR](https://github.com/ethereum/go-ethereum/pull/22461) 

## Types of changes

Please put an x in the boxes related to your change.

- [ ] Bugfix
- [x] New feature or enhancement
- [ ] Others

## Checklist

*Put an x in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.*

- [x] I have read the [CONTRIBUTING GUIDELINES](https://github.com/klaytn/klaytn/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://cla-assistant.io/klaytn/klaytn)
- [x] Lint and unit tests pass locally with my changes (`$ make test`)
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

Please let me know if there is anything lacking in PR or code.

### CODE REVIEW

File: node/doc.go [hunk 1]
```diff
@@ -103,7 +103,7 @@ Functions and variables related to Node are defined in the files listed below
   - node_test.go         : Test functions for testing the Node type
   - service.go           : Defines Service, Service is an individual protocol that can be registered into a node
   - service_test.go      : Functions for testing the Service
-  - utils.go             : Administrative APIs
+  - api.go             : Administrative APIs
```
Comment (ehnuje) [inline node/doc.go:106]: How about aligning the colon with the others?

File: node/api_test.go [hunk 2]
```diff
@@ -0,0 +1,286 @@
+// Modifications Copyright 2018 The klaytn Authors
```
Comment (ehnuje) [inline node/api_test.go:1]:
```suggestion
// Modifications Copyright 2021 The klaytn Authors
```
Isn't it 2021?

File: console/web3ext/web3ext.go [hunk 3]
```diff
@@ -269,6 +269,16 @@ web3._extend({
...
+			params: 5,
+			inputFormatter: [null, null, null, null, null]
+		}),
+		new web3._extend.Method({
+			name: 'stopHTTP',
+			call: 'admin_stopHTTP'
+		}),
 		new web3._extend.Method({
 			name: 'startRPC',
 			call: 'admin_startRPC',
```
Comment (ehnuje) [inline console/web3ext/web3ext.go:284]: Shouldn't below `params` be 5?

**Review by ehnuje — COMMENTED** [review]
LGTM except for the minor comments

File: console/web3ext/web3ext.go [hunk 4]
```diff
@@ -269,6 +269,16 @@ web3._extend({
 			call: 'admin_sleepBlocks',
 			params: 2
 		}),
+		new web3._extend.Method({
+			name: 'startHTTP',
+			call: 'admin_startHTTP',
+			params: 5,
+			inputFormatter: [null, null, null, null, null]
+		}),
+		new web3._extend.Method({
```
Comment (jeongkyun-oh) [inline console/web3ext/web3ext.go:278]: @hyochan-brown Is this method avaliable? The change does not  show the method `stopHTTP`.

File: node/api_test.go [hunk 5]
```diff
@@ -0,0 +1,286 @@
...
+	}
+
+	for _, test := range tests {
+		test := test
+		t.Run(test.name, func(t *testing.T) {
+			t.Parallel()
+
+			// Apply some sane defaults.
+			config := test.cfg
+			// config.Logger = testlog.Logger(t, log.LvlDebug)
```
Comment (jeongkyun-oh) [inline node/api_test.go:180]: Is this comment on purpose?

File: node/api_test.go [hunk 6]
```diff
@@ -0,0 +1,286 @@
...
+			config.P2P.NoDiscovery = true
+
+			// Create Node.
+			stack, err := New(&config)
+			if err != nil {
+				t.Fatal("can't create node:", err)
+			}
+
+			// Register the test config.
+			stack.config.HTTPServerType = "fasthttp"
```
Comment (jeongkyun-oh) [inline node/api_test.go:190]: How about adding `http` type as well?

File: console/web3ext/web3ext.go [hunk 7]
```diff
@@ -269,6 +269,16 @@ web3._extend({
...
+			name: 'stopHTTP',
+			call: 'admin_stopHTTP'
+		}),
 		new web3._extend.Method({
```
Comment (yoomee1313) [inline console/web3ext/web3ext.go:282]: What about adding the comment that startRPC and stopRPC is deprecated as original code does?

File: node/api_test.go (hunk 5 above)
Comment (yoomee1313) [reply node/api_test.go:180]: In geth code, this statement is also commented out.

File: node/doc.go (hunk 1 above)
Comment (hyochan-brown) [reply node/doc.go]: I modified it. Thanks!

File: console/web3ext/web3ext.go (hunk 3 above)
Comment (hyochan-brown) [reply console/web3ext/web3ext.go:285]: `params` is 5 right. Thanks!

File: node/api_test.go (hunk 5 above)
Comment (hyochan-brown) [reply node/api_test.go]: It was in the geth code, but I deleted it because it seemed unnecessary. Thanks!

File: console/web3ext/web3ext.go (hunk 7 above)
Comment (hyochan-brown) [reply console/web3ext/web3ext.go:283]: I added comments. Thanks!

File: node/api_test.go (hunk 6 above)
Comment (hyochan-brown) [reply node/api_test.go]: I added it. Please review the code to see if it is weird.

**Comment by ehnuje** [comment]
@hyochan-brown Could you please check the test failures?

**Review by ehnuje — APPROVED** [review]

File: node/api_test.go [hunk 8]
```diff
@@ -0,0 +1,289 @@
...
+// string/int pointer helpers.
+func sp(s string) *string { return &s }
+func ip(i int) *int       { return &i }
+
+func not(ok bool) string {
+	if ok {
+		return ""
+	}
+	return "not "
+}
```
Comment (yoomee1313) [inline node/api_test.go:289]: What about adding an enter at last?

**Review by yoomee1313 — APPROVED** [review]

**Review by jeongkyun-oh — APPROVED** [review]

**Review by KimKyungup — APPROVED** [review]

**Review by aidan-kwon — APPROVED** [review]

**Pull request merged**